src/ExpressionEvaluator/VisualBasic/Source/ExpressionCompiler/Binders/PlaceholderLocalBinder.vb

```diff
@@ -43,13 +44,17 @@ Namespace Microsoft.CodeAnalysis.VisualBasic.ExpressionEvaluator
                 Return
             End If
 
+            Dim canonicalName = Canonicalize(name)
+
             Dim local As LocalSymbol = Nothing
             If _implicitDeclarations IsNot Nothing Then
-                _implicitDeclarations.TryGetValue(name, local)
+                If Not _implicitDeclarations.TryGetValue(canonicalName, local) Then
```

Comment:
Is this necessary?  (won't local be null if it fails, or are you just being explicit?)